dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -444,19 +444,67 @@ def tensordot(x1, x2, axes=2):
     return call_origin(numpy.tensordot, x1, x2, axes)
 
 
-def vdot(*args, **kwargs):
+def vdot(a, b):
     """
     Return the dot product of two vectors.
 
-    For full documentation refer to :obj:`numpy.vdot`.
+    For full documentation refer to :obj:`numpy.dot`.
+
+    Parameters
+    ----------
+    a : {dpnp_array, usm_ndarray, scalar}
+        First input array. Both inputs `a` and `b` can not be
+        scalars at the same time. If `a` is complex, the complex
+        conjugate is taken before the calculation of the dot product.
+    b : {dpnp_array, usm_ndarray, scalar}
+        Second input array. Both inputs `a` and `b` can not be
+        scalars at the same time.
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        Returns the dot product of `a` and `b`.
 
     See Also
     --------
     :obj:`dpnp.dot` : Returns the dot product.
 
-    Notes
-    -----
-    This function works the same as :obj:`dpnp.dot`.
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> a = np.array([1+2j,3+4j])
+    >>> b = np.array([5+6j,7+8j])
+    >>> np.vdot(a, b)
+    array(70-8j)
+    >>> np.vdot(b, a)
+    array(70+8j)
+
+    Note that higher-dimensional arrays are flattened!
+
+    >>> a = np.array([[1, 4], [5, 6]])
+    >>> b = np.array([[4, 1], [2, 2]])
+    >>> np.vdot(a, b)
+    array(30)
+    >>> np.vdot(b, a)
+    array(30)
+    >>> 1*4 + 4*1 + 5*2 + 6*2
+    30
 
     """
-    return dpnp.dot(*args, **kwargs)
+
+    dpnp.check_supported_arrays_type(a, scalar_type=True)
+    dpnp.check_supported_arrays_type(b, scalar_type=True)
```

Comment:
it can be a single line:
```suggestion
    dpnp.check_supported_arrays_type(a, b, scalar_type=True)
```